### ISSUE TITLE
Better guidance for people who don't understand they need to send LogQL

### ISSUE THREAD

**Issue by kevinburkesegment** [grafana/loki#10414]
For example, I just tried sending the word "segment" to Loki and figured I'd get some logs back with the word "segment" in them, or no logs if nothing matched. Instead I got back:

```
$ curl -v -u "651528:$(cat disk-tokens/grafana-cloudaccesspolicy-token)" https://logs-prod-006.grafana.net/loki/api/v1/query_range\?direction\=BACKWARD\&end\=1693289918339003000\&limit\=30\&query\=segment\&start\=1693286318339003000
> GET /loki/api/v1/query_range?direction=BACKWARD&end=1693289918339003000&limit=30&query=segment&start=1693286318339003000 HTTP/1.1
> Host: logs-prod-006.grafana.net
> Authorization: Basic <redacted> 
> User-Agent: curl/8.2.1
> Accept: */*
>
< HTTP/1.1 400 Bad Request
< Content-Length: 65
< Content-Type: text/plain; charset=utf-8
< Date: Thu, 31 Aug 2023 19:59:09 GMT
< Vary: Accept-Encoding
<
parse error at line 1, col 1: syntax error: unexpected IDENTIFIER
```

Which isn't very helpful at explaining what the problem is or what you should have sent instead. 

I believe this particular error is coming from `pkg/querier/http.go` which calls `http.Error` with the contents of the `syntax.ParseExpr` error. 

So one option would be to return a better error in `ParseExpr`

    expected valid LogQL, got: parse error at line 1, col 1: syntax error: unexpected IDENTIFIER
    parse error at line 1, col 1: syntax error: unexpected IDENTIFIER (LogQL syntax expected)
    
Another option would be to catch the error coming from ParseExpr and rethrow in e.g. `pkg/querier/http.go`, this was my original idea. 

```
HTTP/1.1 400 Bad Request
Content-Length: 65
Content-Type: text/plain; charset=utf-8
Date: Thu, 31 Aug 2023 19:59:09 GMT
Vary: Accept-Encoding
    parse error at line 1, col 1: syntax error: unexpected IDENTIFIER (queries should use LogQL: https://grafana.com/docs/loki/latest/logql/log_queries/)
```

Alternatively, have this endpoint send back JSON as suggested in PR #10413. Would want to see whether logcli could expose the content of an additional field in the error output.

```json
{"status":"error","error":"parse error at line 1, col 1: syntax error: unexpected IDENTIFIER", "more_info": "https://grafana.com/docs/loki/latest/logql/log_queries/"}
```

**Comment by dannykopping** [comment]
Hey @kevinburkesegment 

Sorry, it's been a busy few days. I've added a note to get back to you on this Monday or early next week 👍

**Comment by kevinburkesegment** [comment]
Thank you!

**Comment by dannykopping** [comment]
@kevinburkesegment sincere apologies! This one slipped off my radar.

OK, so there are a couple angles to this particular change you'd like to make:

The one is that we have to reasonably expect customers to read our docs before interacting with the API. We've added https://github.com/grafana/loki/pull/10740 which should clarify that LogQL has to be specified in the query string.

However, if the user doesn't do this and sends an invalid LogQL query we should absolutely inform them in a better way than we do now. To this end, I think it would be a good first step to check if any label selectors have been defined; these are the core of any LogQL query, and with these missing we could send back an error message with a reference to the documentation. The benefit of doing this vs returning an error for any invalid LogQL is that it'd be a more targeted error message and therefore more valuable. What do you think?

As an aside: I'm really interested in improving the _overall_ error messaging experience in Loki. I created a package called [errata](https://github.com/dannykopping/errata/) which I plan to integrate into Loki at some point, when I eventually get control of my infinity backlog 🙃 
If this is something you're interested in, we could discuss some contributions. Maybe the above case is the thin edge of the wedge to get us using errata 🤔

**Comment by till** [comment]
It would be great if the examples in logcli were improved. I ran into this as well while testing. Realized an hour later that I had to send logql to do so. But all the examples read like:

```
      logcli volume
         --timezone=UTC
         --from="2021-01-19T10:00:00Z"
         --to="2021-01-19T20:00:00Z"
         'my-query'
```

**Comment by chaudum** [comment]
I don't think this is a technical problem. It's like not understanding that you need to use SQL to query MySQL or Postgres ... @JStickler updated the docs (see linked PR)
